File: BCC/EDI(CCB0714)/python/fila/model/fila.py
from lista_dupla.model.lista_dupla import ListaDupla

class Fila:
    def __init__(self):
        self.fila = ListaDupla()

    def insereNaFila(self, item):
        self.fila.insereElementoNaLista(self.fila.tail, item)

    def removeDaFila(self) -> None:
        return self.fila.removeElementoDaLista(self.fila.head)

    def isEmpty(self) -> bool:
        return True if self.fila.size == 0 else False
    
    def size(self) -> int:
        return self.fila.size
    
    def mostraFila(self):
        self.fila.percorreListaHeadTail()

File: BCC/EDI(CCB0714)/C/fila/model/fila.c
#include <stdio.h>
#include <stdlib.h>
#include "../../listaEncadeadaSimples/model/listaSimples.c"

//Funções da fila
void insereNaFila(Lista*, int);
int removeDaFila(Lista*);
int filaVazia(Lista*);
int tamanhoFila(Lista*);
void mostraFila(Lista*);

void insereNaFila(Lista* fila, int dado){
    insereElementoNaLista(fila, fila->tail, dado);
}

int removeDaFila(Lista* fila){
//    return removeElementoNaLista(fila, fila->head); // Para a lista dupla
    return removeElementoNaLista(fila, NULL); // Para a lista simples
}

int filaVazia(Lista* fila){
    if (fila->size == 0)
        return 1;
    else
        return 0;
}

int tamanhoFila(Lista* fila){
    return fila->size;
}

void mostraFila(Lista* fila){
    percorreListaHeadTail(fila);
}